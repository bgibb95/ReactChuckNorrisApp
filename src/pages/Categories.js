import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import {
  StyledLink,
  RightContainer,
  LeftContainer,
  MainTitle,
  Page,
  Copy,
} from '../styles/common';
import fetchCategories from '../actions/categories';

export class Categories extends Component {
  componentWillMount() {
    const { fetchCategories: getCategories } = this.props;
    getCategories();
  }

  render() {
    const { categories } = this.props;
    const categoryItems = categories.map((category, index) => (
      // eslint-disable-next-line react/no-array-index-key
      <StyledLink to={`/joke/${category}`} key={index}>
        {category}
      </StyledLink>
    ));
    const { loading } = this.props;

    return (
      <Page>
        <LeftContainer>
          <MainTitle>Joke categories</MainTitle>
          <Copy>
            Select a category to fetch a random Chuck Norris joke, relating to
            your chosen category.
          </Copy>
        </LeftContainer>
        <RightContainer loading={loading}>
          {categoryItems}
        </RightContainer>
      </Page>
    );
  }
}

Categories.propTypes = {
  fetchCategories: PropTypes.func.isRequired,
  categories: PropTypes.arrayOf(PropTypes.string).isRequired,
  loading: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  categories: state.categories.categories,
  loading: state.categories.loading,
});

export default connect(mapStateToProps, { fetchCategories })(Categories);
