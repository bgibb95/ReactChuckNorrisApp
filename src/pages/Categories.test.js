import React from 'react';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Categories } from './Categories';

describe('Categories page', () => {
  const props = {
    fetchCategories: jest.fn(),
    categories: ['dev'],
    loading: false,
  };

  test('It matches the snapshot', () => {
    const component = create(<MemoryRouter><Categories {...props} /></MemoryRouter>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Should call getCategories on mount', () => {
    expect(props.fetchCategories).toHaveBeenCalled();
  });
});
