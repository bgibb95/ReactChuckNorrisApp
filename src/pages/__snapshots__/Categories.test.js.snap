// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories page It matches the snapshot 1`] = `
<div
  className="common__Page-sc-4zl79x-0 tfEnI"
>
  <div
    className="common__Container-sc-4zl79x-6 common__LeftContainer-sc-4zl79x-8 UmaQz"
  >
    <h1
      className="common__MainTitle-sc-4zl79x-5 WItnH"
    >
      Joke categories
    </h1>
    <p
      className="common__Copy-sc-4zl79x-1 iOKRhW"
    >
      Select a category to fetch a random Chuck Norris joke, relating to your chosen category.
    </p>
  </div>
  <div
    className="common__Container-sc-4zl79x-6 common__RightContainer-sc-4zl79x-7 hVahFF"
  >
    <a
      className="common__StyledLink-sc-4zl79x-3 eVeMck"
      href="/joke/dev"
      onClick={[Function]}
    >
      dev
    </a>
  </div>
</div>
`;
